# 1. Importing libraries and Dataset

In [1]:
import pandas as pd 
import numpy as np
from nltk.tokenize import sent_tokenize,word_tokenize
from nltk.stem import WordNetLemmatizer
import re
from sklearn.decomposition import LatentDirichletAllocation
from nltk.corpus import stopwords
from sklearn.model_selection import GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import mean_absolute_error,mean_squared_error
from sklearn.feature_extraction.text import CountVectorizer
import gensim
import pyLDAvis.gensim
import matplotlib.pyplot as plt
from collections import Counter
import warnings

In [2]:
%matplotlib inline
warnings.filterwarnings('ignore')

In [3]:
data_df=pd.read_csv("nips-papers/papers.csv")

In [4]:
data_df.columns

Index(['id', 'year', 'title', 'event_type', 'pdf_name', 'abstract',
       'paper_text'],
      dtype='object')

In [5]:
data_df['paper_text'].head().apply(len)

0    21643
1    15505
2    20523
3    19441
4    20219
Name: paper_text, dtype: int64

In [6]:
author_name =pd.read_csv('nips-papers/authors.csv')

In [7]:
author_name.head()

id                  name
0      1        Hisashi Suzuki
1     10           David Brady
2    100  Santosh S. Venkatesh
3   1000     Charles Fefferman
4  10000         Artur Speiser

# 2.Text Trocessing

In [8]:
data_df[['title','pdf_name']].head(10)

title  \
0  Self-Organization of Associative Database and ...   
1  A Mean Field Theory of Layer IV of Visual Cort...   
2  Storing Covariance by the Associative Long-Ter...   
3  Bayesian Query Construction for Neural Network...   
4  Neural Network Ensembles, Cross Validation, an...   
5  Using a neural net to instantiate a deformable...   
6           Plasticity-Mediated Competitive Learning   
7  ICEG Morphology Classification using an Analog...   
8  Real-Time Control of a Tokamak Plasma Using Ne...   
9  Pulsestream Synapses with Non-Volatile Analogu...   

                                            pdf_name  
0  1-self-organization-of-associative-database-an...  
1  10-a-mean-field-theory-of-layer-iv-of-visual-c...  
2  100-storing-covariance-by-the-associative-long...  
3  1000-bayesian-query-construction-for-neural-ne...  
4  1001-neural-network-ensembles-cross-validation...  
5  1002-using-a-neural-net-to-instantiate-a-defor...  
6  1003-plasticity-mediated-competitive-learning.pdf  
7  1004-iceg-morphology-classification-using-an-a...  
8  1005-real-time-control-of-a-tokamak-plasma-usi...  
9  1006-pulsestream-synapses-with-non-volatile-an...

Since title and pdf_name is  same hence removing pdf_name 

In [9]:
data_df['event_type'].unique()

array([nan, 'Oral', 'Spotlight', 'Poster'], dtype=object)

In [10]:
data_df.drop(['pdf_name','event_type'],axis=1,inplace=True)

Since our main object is to summarize the text hence and tagging the key words hence we donot require the event type 

In [11]:
data_df.head()

id  year                                              title  \
0     1  1987  Self-Organization of Associative Database and ...   
1    10  1987  A Mean Field Theory of Layer IV of Visual Cort...   
2   100  1988  Storing Covariance by the Associative Long-Ter...   
3  1000  1994  Bayesian Query Construction for Neural Network...   
4  1001  1994  Neural Network Ensembles, Cross Validation, an...   

           abstract                                         paper_text  
0  Abstract Missing  767\n\nSELF-ORGANIZATION OF ASSOCIATIVE DATABA...  
1  Abstract Missing  683\n\nA MEAN FIELD THEORY OF LAYER IV OF VISU...  
2  Abstract Missing  394\n\nSTORING COVARIANCE BY THE ASSOCIATIVE\n...  
3  Abstract Missing  Bayesian Query Construction for Neural\nNetwor...  
4  Abstract Missing  Neural Network Ensembles, Cross\nValidation, a...

In [12]:
paper_separate_abstract = data_df[(data_df['abstract']!='Abstract Missing')].index

In [13]:
paper_separate_abstract

Int64Index([ 941, 1067, 2384, 2385, 2388, 2389, 2390, 2393, 2394, 2396,
            ...
            6937, 6938, 6939, 6940, 6941, 6943, 6944, 6945, 6946, 6947],
           dtype='int64', length=3924)

In [14]:
len(data_df[(data_df['abstract']!='Abstract Missing')].index)

3924

In [15]:
print(data_df['abstract'].iloc[941])

Non-negative matrix factorization (NMF) has previously been shown to 
be a useful decomposition for multivariate data. Two different multi- 
plicative algorithms for NMF are analyzed. They differ only slightly in 
the multiplicative factor used in the update rules. One algorithm can be 
shown to minimize the conventional least squares error while the other 
minimizes the generalized Kullback-Leibler divergence. The monotonic 
convergence of both algorithms can be proven using an auxiliary func- 
tion analogous to that used for proving convergence of the Expectation- 
Maximization algorithm. The algorithms can also be interpreted as diag- 
onally rescaled gradient descent, where the rescaling factor is optimally 
chosen to ensure convergence. 


In [16]:
data_df = data_df.iloc[paper_separate_abstract]

In [17]:
data_df.drop('paper_text',axis=1,inplace=True)

Since abstract is present in 3924 text hence removing the full papers of this journals

In [18]:
data_df.reset_index(inplace = True)

In [19]:
data_df.drop('index',axis=1,inplace= True)

In [20]:
print(data_df['abstract'][0])

Non-negative matrix factorization (NMF) has previously been shown to 
be a useful decomposition for multivariate data. Two different multi- 
plicative algorithms for NMF are analyzed. They differ only slightly in 
the multiplicative factor used in the update rules. One algorithm can be 
shown to minimize the conventional least squares error while the other 
minimizes the generalized Kullback-Leibler divergence. The monotonic 
convergence of both algorithms can be proven using an auxiliary func- 
tion analogous to that used for proving convergence of the Expectation- 
Maximization algorithm. The algorithms can also be interpreted as diag- 
onally rescaled gradient descent, where the rescaling factor is optimally 
chosen to ensure convergence. 


Having the word count 

In [21]:
data_df.head()

id  year                                              title  \
0  1861  2000   Algorithms for Non-negative Matrix Factorization   
1  1975  2001  Characterizing Neural Gain Control using Spike...   
2  3163  2007                        Competition Adds Complexity   
3  3164  2007  Efficient Principled Learning of Thin Junction...   
4  3167  2007     Regularized Boost for Semi-Supervised Learning   

                                            abstract  
0  Non-negative matrix factorization (NMF) has pr...  
1  Spike-triggered averaging techniques are effec...  
2  It is known that determinining whether a DEC-P...  
3  We present the first truly polynomial algorith...  
4  Semi-supervised inductive learning concerns ho...

In [22]:
def text_processing(df,col):
    temp_df = df[col]
    # 1.Remove punctuation
    temp_df = temp_df.apply(lambda x: re.sub('[[^a-zA-Z]]',' ',x))
    # 2. converting lower case
    temp_df = temp_df.apply(lambda x: x.lower())
    # 3. removing special character and digit
    temp_df = temp_df.apply(lambda x: re.sub("(\\d|\\W)+"," ",x))
    return temp_df

In [23]:
data_df['abstract'] =text_processing(data_df,'abstract')

In [25]:
print(data_df['abstract'][0])

non negative matrix factorization nmf has previously been shown to be a useful decomposition for multivariate data two different multi plicative algorithms for nmf are analyzed they differ only slightly in the multiplicative factor used in the update rules one algorithm can be shown to minimize the conventional least squares error while the other minimizes the generalized kullback leibler divergence the monotonic convergence of both algorithms can be proven using an auxiliary func tion analogous to that used for proving convergence of the expectation maximization algorithm the algorithms can also be interpreted as diag onally rescaled gradient descent where the rescaling factor is optimally chosen to ensure convergence 


In [26]:
def tokenize_lemmatize(df,col):
    temp_df =df[col]
    #1. Word Tokenization:
    temp_df = temp_df.apply(lambda x : word_tokenize(x))
    word_no_pre = temp_df.apply(lambda x: len(x))
    temp_df = temp_df.apply(lambda x : [i for i in x if not i in stopwords.words('english')])
    #2. Word Lemmatization:
    lemmatize =WordNetLemmatizer()
    temp_df = temp_df.apply(lambda x: [lemmatize.lemmatize(i) for i in x])
    word_no_post =temp_df.apply(lambda x: len(x))
    temp_df = temp_df.apply(lambda x: " ".join(x))
    return temp_df,word_no_pre,word_no_post

In [27]:
data_df['abstract_post'],data_df['word_count_pre'],data_df['word_count_post']=tokenize_lemmatize(data_df,'abstract')

In [28]:
data_df.head()

id  year                                              title  \
0  1861  2000   Algorithms for Non-negative Matrix Factorization   
1  1975  2001  Characterizing Neural Gain Control using Spike...   
2  3163  2007                        Competition Adds Complexity   
3  3164  2007  Efficient Principled Learning of Thin Junction...   
4  3167  2007     Regularized Boost for Semi-Supervised Learning   

                                            abstract  \
0  non negative matrix factorization nmf has prev...   
1  spike triggered averaging techniques are effec...   
2  it is known that determinining whether a dec p...   
3  we present the first truly polynomial algorith...   
4  semi supervised inductive learning concerns ho...   

                                       abstract_post  word_count_pre  \
0  non negative matrix factorization nmf previous...             108   
1  spike triggered averaging technique effective ...              83   
2  known determinining whether dec pomdp namely c...              70   
3  present first truly polynomial algorithm learn...             144   
4  semi supervised inductive learning concern lea...             123   

   word_count_post  
0               67  
1               52  
2               40  
3               89  
4               81

In [29]:
 token_df = data_df['abstract_post'].apply(lambda x:x.split(" "))

In [30]:
# creating word dictionary
dictionary = gensim.corpora.Dictionary(token_df)
#converting dictionary into a bag of words 
word_map =[dictionary.doc2bow(text) for text in token_df]
# Build LDA model
lda_model = gensim.models.ldamodel.LdaModel(corpus=word_map,
                                           id2word=dictionary,
                                           num_topics=4, 
                                           random_state=100,
                                           update_every=1,
                                           chunksize=100,
                                           passes=50,
                                           per_word_topics=True)

In [30]:
pyLDAvis.enable_notebook()
pyLDAvis.gensim.prepare(lda_model, word_map, dictionary)

PreparedData(topic_coordinates=              x         y  topics  cluster       Freq
topic                                                
3      0.096551  0.119806       1        1  29.222599
1     -0.033844 -0.159776       2        1  27.591478
2     -0.232030  0.059157       3        1  22.493319
0      0.169324 -0.019186       4        1  20.692600, topic_info=     Category         Freq           Term        Total  loglift  logprob
term                                                                    
785   Default  2491.000000        network  2491.000000  30.0000  30.0000
156   Default  4864.000000          model  4864.000000  29.0000  29.0000
80    Default  1214.000000         neural  1214.000000  28.0000  28.0000
323   Default  1106.000000          image  1106.000000  27.0000  27.0000
27    Default  1027.000000         matrix  1027.000000  26.0000  26.0000
0     Default  4097.000000      algorithm  4097.000000  25.0000  25.0000
560   Default  1067.000000          bound  1067.000000  24.0000  24.0000
974   Default   876.000000           deep   876.000000  23.0000  23.0000
146   Default   924.000000      inference   924.000000  22.0000  22.0000
540   Default   725.000000         kernel   725.000000  21.0000  21.0000
595   Default   871.000000          graph   871.000000  20.0000  20.0000
482   Default  3034.000000        problem  3034.000000  19.0000  19.0000
8     Default  3151.000000           data  3151.000000  18.0000  18.0000
633   Default   562.000000      estimator   562.000000  17.0000  17.0000
407   Default  1745.000000       function  1745.000000  16.0000  16.0000
486   Default   497.000000         sparse   497.000000  15.0000  15.0000
1135  Default   605.000000         latent   605.000000  14.0000  14.0000
603   Default   808.000000              n   808.000000  13.0000  13.0000
155   Default  3412.000000       learning  3412.000000  12.0000  12.0000
221   Default  1019.000000   optimization  1019.000000  11.0000  11.0000
755   Default   445.000000     regression   445.000000  10.0000  10.0000
22    Default   966.000000       gradient   966.000000   9.0000   9.0000
77    Default  1010.000000         linear  1010.000000   8.0000   8.0000
233   Default  1471.000000           task  1471.000000   7.0000   7.0000
875   Default   444.000000   architecture   444.000000   6.0000   6.0000
1032  Default   560.000000       bayesian   560.000000   5.0000   5.0000
234   Default  1033.000000       training  1033.000000   4.0000   4.0000
786   Default   767.000000        optimal   767.000000   3.0000   3.0000
1714  Default   524.000000           loss   524.000000   2.0000   2.0000
291   Default   406.000000              x   406.000000   1.0000   1.0000
...       ...          ...            ...          ...      ...      ...
4718   Topic4    92.071892     completion    92.792290   1.5676  -6.7106
2493   Topic4   331.801910           rank   346.182831   1.5330  -5.4286
631    Topic4   427.881348    dimensional   535.840942   1.3504  -5.1743
465    Topic4   448.968414     estimation   590.261169   1.3018  -5.1262
690    Topic4   255.972748      dimension   309.458984   1.3856  -5.6881
603    Topic4   557.567871              n   808.160217   1.2042  -4.9095
9      Topic4   168.094971  decomposition   190.016388   1.4528  -6.1086
330    Topic4   354.418732            low   484.462799   1.2628  -5.3627
77     Topic4   629.499146         linear  1010.460510   1.1022  -4.7882
57     Topic4   530.407288       analysis   866.958374   1.0840  -4.9595
539    Topic4   455.221680              k   737.254883   1.0932  -5.1123
844    Topic4   304.020996         vector   441.216125   1.2030  -5.5160
78     Topic4  1176.537354         method  3190.352783   0.5778  -4.1628
441    Topic4   273.308380    statistical   409.288025   1.1716  -5.6225
542    Topic4   259.436951           mean   378.519958   1.1976  -5.6746
320    Topic4   393.357849           high   759.663452   0.9172  -5.2584
482    Topic4   796.346497        problem  3034.625000  

In [31]:
topics = lda_model.show_topics(formatted=False)

In [32]:
topix={}
for i in range(len(topics)):
    val=[]
    print( 'Topic '+str(i))
    for j in range(len(topics[i][1])):
        val.append(topics[i][1][j][0])
    print(val)
    topix['Topic '+str(i)]=val

Topic 0
['method', 'matrix', 'problem', 'data', 'kernel', 'algorithm', 'linear', 'estimator', 'n', 'analysis']
Topic 1
['model', 'data', 'approach', 'learning', 'method', 'algorithm', 'inference', 'graph', 'distribution', 'based']
Topic 2
['network', 'model', 'neural', 'image', 'deep', 'task', 'training', 'learning', 'representation', 'object']
Topic 3
['algorithm', 'learning', 'problem', 'function', 'bound', 'show', 'optimization', 'gradient', 'optimal', 'result']


In [33]:
#len(lda_model[word_map])
len(dictionary)

12641

In [54]:
no_feat =5000
# LDA can only use raw term counts  because it is a probabilistic graphical model
tf_vectorizer = CountVectorizer(max_features=no_feat,stop_words='english')
X_tf = tf_vectorizer.fit_transform(data_df['abstract_post'])
tf_feat_name = tf_vectorizer.get_feature_names()

lda = LatentDirichletAllocation(learning_method='online')
lda_param = {'n_components': [4,5,6], 'learning_decay': [0.5,0.7,0.9]}
lda_model = GridSearchCV(estimator=lda,param_grid=lda_param)
lda_model.fit(X_tf)

GridSearchCV(cv=None, error_score='raise',
       estimator=LatentDirichletAllocation(batch_size=128, doc_topic_prior=None,
             evaluate_every=-1, learning_decay=0.7,
             learning_method='online', learning_offset=10.0,
             max_doc_update_iter=100, max_iter=10, mean_change_tol=0.001,
             n_components=10, n_jobs=1, n_topics=None, perp_tol=0.1,
             random_state=None, topic_word_prior=None,
             total_samples=1000000.0, verbose=0),
       fit_params=None, iid=True, n_jobs=1,
       param_grid={'n_components': [4, 5, 6], 'learning_decay': [0.5, 0.7, 0.9]},
       pre_dispatch='2*n_jobs', refit=True, return_train_score='warn',
       scoring=None, verbose=0)

In [55]:
lda_model.best_params_

{'learning_decay': 0.9, 'n_components': 4}

In [56]:
best_lda_model =lda_model.best_estimator_

In [57]:
lda_output = best_lda_model.fit_transform(X_tf)

In [58]:
np.argsort(best_lda_model.components_[1])[:-10-1:-1]

array([2855, 2973, 2975, 2147, 2526, 4534, 3045, 1146, 4331, 1747])

In [59]:
def visualizing_topic_cluster(model,stop_len,feat_name):
    topic={}
    for index,topix in enumerate(model.components_):
        topic[index]= [feat_name[i] for i in topix.argsort()[:-stop_len-1:-1]]
    
    return topic
        

In [60]:
dict_topic=visualizing_topic_cluster(best_lda_model,stop_len=10,feat_name=tf_feat_name)

In [61]:
len(best_lda_model.components_)

4

In [62]:
len([print('Topic'+str(key),dict_topic[key]) for key in dict_topic])

Topic0 ['method', 'data', 'problem', 'algorithm', 'learning', 'model', 'matrix', 'approach', 'function', 'kernel']
Topic1 ['model', 'network', 'neural', 'image', 'learning', 'task', 'object', 'deep', 'state', 'feature']
Topic2 ['model', 'data', 'inference', 'learning', 'distribution', 'latent', 'label', 'approach', 'method', 'variable']
Topic3 ['algorithm', 'problem', 'learning', 'function', 'bound', 'optimal', 'result', 'time', 'policy', 'stochastic']


4

In [63]:
columns=['Topic'+ i for i in list(map(str,list(dict_topic.keys())))]

In [64]:
lda_df = pd.concat(objs=[data_df,pd.DataFrame(lda_output,columns=columns).apply(lambda x : np.round(x,2)),pd.DataFrame([[np.argmax(x),dict_topic[np.argmax(x)]] for x in lda_output],columns=['Major_topic','keywords'])],axis=1)

In [65]:
columns

['Topic0', 'Topic1', 'Topic2', 'Topic3']

In [66]:
def rmse_calculation(x,y):
    x,y =x,y
    error=[]
    len_y =len(y)
    for i in range(len_y):
        error.append(np.sqrt(np.power((np.array(x)-np.array(y[i])),2).sum()))
    return error 

In [102]:
def reccomendation(df,paper_id,col):
    paper_1 = df[(df['id']==paper_id)]
    dominant_topic = int(paper_1['Major_topic'])
    paper_2 = df[(df['Major_topic']== dominant_topic)]
    paper_2 = paper_2.drop(paper_1.index)
    x= paper_1[columns].values.tolist()
    y= paper_2[columns].values.tolist()
    error = np.round(rmse_calculation(x,y),2)
    paper_2['error'] = error
    return  paper_2[['id','title','error']]

In [156]:
lda_df

id  year                                              title  \
0     1861  2000   Algorithms for Non-negative Matrix Factorization   
1     1975  2001  Characterizing Neural Gain Control using Spike...   
2     3163  2007                        Competition Adds Complexity   
3     3164  2007  Efficient Principled Learning of Thin Junction...   
4     3167  2007     Regularized Boost for Semi-Supervised Learning   
5     3168  2007  Simplified Rules and Theoretical Analysis for ...   
6     3169  2007  Predicting human gaze using low-level saliency...   
7     3171  2007        Mining Internet-Scale Software Repositories   
8     3172  2007        Continuous Time Particle Filtering for fMRI   
9     3174  2007  An online Hebbian learning rule that performs ...   
10    3176  2007              Discriminative K-means for Clustering   
11    3178  2007  The Epoch-Greedy Algorithm for Multi-armed Ban...   
12    3186  2007  Local Algorithms for Approximate Inference in ...   
13    3188  2007                  Catching Change-points with Lasso   
14    3190  2007  Evaluating Search Engines by Modeling the Rela...   
15    3191  2007           Random Projections for Manifold Learning   
16    3195  2007  Learning the structure of manifolds using rand...   
17    3196  2007        A Probabilistic Approach to Language Change   
18    3197  2007  Online Linear Regression and Its Application t...   
19    3199  2007          Scan Strategies for Meteorological Radars   
20    3200  2007  Fixing Max-Product: Convergent Message Passing...   
21    3202  2007  Parallelizing Support Vector Machines on Distr...   
22    3203  2007                 Predictive Matrix-Variate t Models   
23    3204  2007  Modelling motion primitives and their timing i...   
24    3206  2007                Learning the 2-D Topology of Images   
25    3209  2007              On higher-order perceptron algorithms   
26    3210  2007  Configuration Estimates Improve Pedestrian Fin...   
27    3211  2007  Using Deep Belief Nets to Learn Covariance Ker...   
28    3212  2007              Learning Bounds for Domain Adaptation   
29    3213  2007  Unconstrained On-line Handwriting Recognition ...   
...    ...   ...                                                ...   
3894  7255  2017  Attend and Predict: Understanding Gene Regulat...   
3895  7256  2017  Acceleration and Averaging in Stochastic Desce...   
3896  7257  2017      Kernel functions based on triplet comparisons   
3897  7258  2017  An Error Detection and Correction Framework fo...   
3898  7259  2017  Style Transfer from Non-Parallel Text by Cross...   
3899  7260  2017                     Cross-Spectral Factor Analysis   
3900  7261  2017  Stochastic Submodular Maximization: The Case o...   
3901  7262  2017  Affinity Clustering: Hierarchical Clustering a...   
3902  7263  2017  Unsupervised Transformation Learning via Conve...   
3903  7264  2017  A Sharp Error Analysis for the Fused Lasso, wi...   
3904  7265  2017  Linear Time Computation of Moments in Sum-Prod...   
3905  7266  2017  A Meta-Learning Perspective on Cold-Start Reco...   
3906  7267  2017  Predicting Scene Parsing and Motion Dynamics i...   
3907  7268  2017  Sticking the Landing: Simple, Lower-Variance G...   
3908  7269  2017  Efficient Approximation Algorithms for Strings...   
3909  7270  2017  Kernel Feature Selection via Conditional Covar...   
3910  7271  2017  Convergence of Gradient EM on Multi-component ...   
3911  7272  2017  Real Time Image Saliency for Black Box Classif...   
3912  7273  2017  Houdini: Fooling Deep Structured Visual and Sp...   
3913  7274  2017  Efficient and Flexible Inference for Stochasti...   
3914  7275  2017  When Cyclic Coordinate Descent Outperforms Ran...   
3915  7276  2017                         Active Learning from Peers   
3916  7277  2017  Experimental Design for Learning Causal Graphs...   
3917  7278  2017                         Learning to Model the Tail   
3918  7279  2017  Stochastic Mir

In [157]:
reccomended =reccomendation(lda_df,3167,columns)

In [158]:
reccomended.sort_values('error')

id                                              title  error
3433  6794  Consistent Multitask Learning with Nonlinear O...   0.00
3361  6722  Fixed-Rank Approximation of a Positive-Semidef...   0.00
690   4000  Sufficient Conditions for Generating Group Lev...   0.00
1877  5237                     Learning with Fredholm Kernels   0.00
678   3988  Efficient and Robust Feature Selection via Joi...   0.00
1879  5239      Kernel Mean Estimation via Spectral Filtering   0.00
3376  6737  Generalized Linear Model Regression under Dist...   0.00
1880  5240      Subspace Embeddings for the Polynomial Kernel   0.00
1284  4626  Exact and Stable Recovery of Sequences of Sign...   0.00
655   3965    Network Flow Algorithms for Structured Sparsity   0.00
644   3954  Learning Kernels with Radiuses of Minimum Encl...   0.00
1288  4630  Compressive Sensing MRI with Wavelet Tree Spar...   0.00
2474  5834  Structured Estimation with Atomic Norms: Gener...   0.00
3390  6751           Linear regression without correspondence   0.00
626   3936  Relaxed Clipping: A Global Training Method for...   0.00
2978  6338  Stein Variational Gradient Descent: A General ...   0.00
1566  4923  BIG & QUIC: Sparse Inverse Covariance Estimati...   0.00
2469  5829  HONOR: Hybrid Optimization for NOn-convex Regu...   0.00
1274  4615      Matrix reconstruction with the local max norm   0.00
716   4026  Random Conic Pursuit for Semidefinite Programming   0.00
3319  6680                 A New Theory for Matrix Completion   0.00
729   4039                  Random Projection Trees Revisited   0.00
1773  5130  Robust Transfer Principal Component Analysis w...   0.00
2541  5901                 Sparse PCA via Bipartite Matchings   0.00
3262  6623      Parametric Simplex Method for Sparse Learning   0.00
2997  6357  High-Rank Matrix Completion and Clustering und...   0.00
799   4110  An Inverse Power Method for Nonlinear Eigenpro...   0.00
3263  6624                      Group Sparse Additive Machine   0.00
788   4099  Sparse Inverse Covariance Selection via Altern...   0.00
785   4096  The LASSO risk: asymptotic results and real wo...   0.00
...    ...                                                ...    ...
386   3691                                Group Sparse Coding   0.72
3499  6860     Dual Discriminator Generative Adversarial Nets   0.72
1867  5227  Factoring Variations in Natural Images with De...   0.72
3573  6934  Bayesian Inference of Individualized Treatment...   0.72
3338  6699        Learning to Pivot with Adversarial Networks   0.72
2603  5963                          Rectified Factor Networks   0.72
2729  6089                     Graphons, mergeons, and so on!   0.72
3602  6963  Joint distribution optimal transportation for ...   0.73
1233  4568  Dynamic Pruning of Factor Graphs for Maximum M...   0.73
3739  7100  Excess Risk Bounds for the Bayes Risk using Va...   0.73
1267  4608  A systematic approach to extracting semantic i...   0.73
3022  6382  Leveraging Sparsity for Efficient Submodular D...   0.73
3719  7080                      Maximum Margin Interval Trees   0.73
3843  7204  Spectrally-normalized margin bounds for neural...   0.73
1602  4959                   Marginals-to-Models Reducibility   0.74
2329  5689          Rate-Agnostic (Causal) Structure Learning   0.74
1287  4629  Distributed Probabilistic Learning for Camera ...   0.75
2405  5765                                   The Human Kernel   0.75
2093  5453                           (Almost) No Label No Cry   0.75
3454  6815  MMD GAN: Towards Deeper Understanding of Momen...   0.75
2036  5396  Distributed Power-law Graph Computing: Theoret...   0.75
2318  5678  Local Expectation Gradients for Black Box Vari...   0.76
1559  4916  Approximate Inference in Continuous Determinan...   0.76
3334  6695                         Learning Overcomplete HMMs   0.76
2757  6117  Bayesian Optimization with Robust Bayesian Neu...   0.76
1858  5215  Reservoir Boosting : Between Online and Offlin...   0.76
3744  

In [168]:
lda_df.iloc[4]['abstract']

'semi supervised inductive learning concerns how to learn a decision rule from a data set containing both labeled and unlabeled data several boosting algorithms have been extended to semi supervised learning with various strategies to our knowledge however none of them takes local smoothness constraints among data into account during ensemble learning in this paper we introduce a local smoothness regularizer to semi supervised boosting algorithms based on the universal optimization framework of margin cost functionals our regularizer is applicable to existing semi supervised boosting algorithms to improve their generalization and speed up their training comparative results on synthetic benchmark and real world tasks demonstrate the effectiveness of our local smoothness regularizer we discuss relevant issues and relate our regularizer to previous work '

In [172]:
lda_df.iloc[1877]['abstract']

'in this paper we propose a framework for supervised and semi supervised learning based on reformulating the learning problem as a regularized fredholm integral equation our approach fits naturally into the kernel framework and can be interpreted as constructing new data dependent kernels which we call fredholm kernels we proceed to discuss the noise assumption for semi supervised learning and provide evidence evidence both theoretical and experimental that fredholm kernels can effectively utilize unlabeled data under the noise assumption we demonstrate that methods based on fredholm learning show very competitive performance in the standard semi supervised learning setting '